src/test/java/uk/ac/ox/oxfish/fisher/strategies/destination/fad/FadDeploymentRouteSelectorTest.java

```diff
@@ -137,36 +139,42 @@ public void test() {
         final ImmutableMap<Route, Double> routeValues =
             routeSelector.evaluateRoutes(fisher, possibleRoutes, 0)
                 .collect(toImmutableMap(Entry::getKey, Entry::getValue));
-        assertEquals(ImmutableMap.of(shortRoute, 3.0, longRoute, 4.0), routeValues);
 
-        final MersenneTwisterFast rng = new MersenneTwisterFast();
+        assertEquals(1, routeValues.entrySet().stream().filter(entry -> entry.getKey().isSameAs(shortRoute) && entry.getValue() == 3.0).count());
+        assertEquals(1, routeValues.entrySet().stream().filter(entry -> entry.getKey().isSameAs(longRoute) && entry.getValue() == 4.0).count());
 
-        final Map<Route, Long> routeSelectionCounts = Stream
+        final Map<Boolean, Long> routeSelectionCounts = Stream
             .generate(() -> routeSelector.selectRoute(fisher, 0, rng))
             .flatMap(Streams::stream)
             .limit(100)
-            .collect(groupingBy(identity(), counting()));
+            .collect(groupingBy(r -> r.isSameAs(shortRoute), counting()));
 
         // with no travel costs the shortest route should be picked roughly 43% of the time
+        final double shortRouteValue =
+            routeValues.entrySet().stream()
+                .filter(entry -> entry.getKey().isSameAs(shortRoute))
+                .findFirst().map(Entry::getValue).get();
+        final long shortRouteSelectionCount = routeSelectionCounts.get(true);
+
         assertEquals(
```

Comment:
How come this fails from time to time? 
Is it a matter of the limit being too low?

Reply:
Yeah, I'm taking a sample of 100 and using a delta of 0.1 in the comparison. I'll try to increase the sample size and see if it prevents it from failing so often. I'm also thinking that perhaps it's not such a good thing to use probabilistic tests, as they tend to raise false alarms (or slow things down a lot for large samples...)